pkg/webservice/handlers.go

```diff
@@ -527,3 +527,45 @@ func updateConfiguration(conf string) (string, error) {
 	}
 	return "", fmt.Errorf("config plugin not found")
 }
+
+func getPartitions(w http.ResponseWriter, r *http.Request) {
+	writeHeaders(w)
+
+	lists := gClusterInfo.ListPartitions()
+	for _, k := range lists {
+		partitionInfo := getPartition(k)
+
+		if err := json.NewEncoder(w).Encode(partitionInfo); err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+		}
+	}
+}
+
+func getPartition(name string) *dao.PartitionInfo {
+	partitionInfo := &dao.PartitionInfo{}
+	partitionContext := gClusterInfo.GetPartition(name)
+	partitionInfo.Name = partitionContext.Name
+	partitionInfo.State = partitionContext.StateMachine.Current()
+	partitionInfo.LastStateTransitionTime = partitionContext.StateTime.String()
+
+	capacityInfo := dao.PartitionCapacity{}
+	capacityInfo.Capacity = partitionContext.GetTotalPartitionResource().String()
+	capacityInfo.UsedCapacity = partitionContext.Root.GetAllocatedResource().String()
+	partitionInfo.Capacity = capacityInfo
+	partitionInfo.NodeSortingPolicy = partitionContext.GetNodeSortingPolicy().String()
+
+	applicationsInfo := dao.Applications{}
+	appList := partitionContext.GetApplications()
+	applicationsState := make(map[string]int)
+
+	for _, app := range appList {
+		applicationsState[app.GetApplicationState()]++
+	}
+	applicationsInfo.Running = applicationsState["Running"]
```

Comment:
could we optimize this code a bit. too much of impl related to comparison etc.

Reply:
Can you please check the recent changes? It has been taken care.

Reply:
hi @manirajv06  for this part. Can we avoid using hardcoded states for Applications struct.
Maybe we can change it to

```
type PartitionInfo struct {
   ...
   appCounts map[string]int
}
```
then we can simply aggregate the count of apps by their state. What do you think?

Reply:
hi @yangwwei  Sorry for the delay. 

applicationsState in handlers#getPartition() is a map contains the state wise count. 

applicationsInfo ( dao.Applications{} ) in handlers#getPartition() has been used to display the state wise count (after aggregating some app states. For ex, Killed and Rejected as Failed) in JSON output.

I am not sure how defining appCounts map[string]int in PartitionInfo struct helps in this case as we are manipulating some app states. Please correct my understanding if needed.